### PR TITLE
Improve types

### PR DESCRIPTION
**Pull request by axelhzf**
I tried to address some problems mentioned on #20

I also include a new example using axios, with a `useRequest` function like the one implemented on https://github.com/zeit/swr/blob/master/examples/axios-typescript/libs/useRequest.ts

### CODE REVIEW

**Comment by darrenjennings** [comment]
This looks good to me, though @axelhzf can you rename the example folder to be `axios-typescript-nuxt`? 

@blocka does this meet the requirements from https://github.com/Kong/swrv/issues/19#issuecomment-603628151?

File: src/types.ts [hunk 1]
```diff
@@ -14,12 +16,13 @@ export interface IConfig {
   ) => void
 }
 
-export interface IResponse {
-  data?: Ref<any>
-  error?: Ref<any>
+export interface IResponse<Data = any, Error = any> {
+  data?: Ref<Data>
```
Comment (blocka) [inline src/types.ts:20]: Can this be ```data: Ref<Data | undefined>``` instead?

Comment (axelhzf) [reply src/types.ts]: Updated, thanks

**Comment by blocka** [comment]
After the last push, I think it addresses all my issues.

**Comment by darrenjennings** [comment]
These typings helped quite a bit and even found a bug internally. Thanks you @axelhzf, you can fill out this form https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#contributor-t-shirt if you are interested in a contributor shirt.

**Pull request merged**